production/ksonnet/loki/ingester.libsonnet

```diff
@@ -4,22 +4,26 @@ local k = import 'ksonnet-util/kausal.libsonnet';
   local pvc = k.core.v1.persistentVolumeClaim,
   local volumeMount = k.core.v1.volumeMount,
   local statefulSet = k.apps.v1.statefulSet,
+  local podDisruptionBudget = k.policy.v1.podDisruptionBudget,
+
+  local name = 'ingester',
 
   // The ingesters should persist TSDB blocks and WAL on a persistent
   // volume in order to be crash resilient.
   local ingester_data_pvc =
-    pvc.new() +
+    pvc.new('ingester-data') +
     pvc.mixin.spec.resources.withRequests({ storage: $._config.ingester_data_disk_size }) +
     pvc.mixin.spec.withAccessModes(['ReadWriteOnce']) +
-    pvc.mixin.spec.withStorageClassName($._config.ingester_data_disk_class) +
-    pvc.mixin.metadata.withName('ingester-data'),
+    pvc.mixin.spec.withStorageClassName($._config.ingester_data_disk_class),
 
-  newIngesterStatefulSet(name, container, with_anti_affinity=true)::
-    // local ingesterContainer = container + $.core.v1.container.withVolumeMountsMixin([
-    //   volumeMount.new('ingester-data', '/data'),
-    // ]);
+  local ingester_wal_pvc =
+    pvc.new('ingester-wal') +
+    pvc.mixin.spec.resources.withRequests({ storage: $._config.ingester_wal_disk_size }) +
+    pvc.mixin.spec.withAccessModes(['ReadWriteOnce']) +
+    pvc.mixin.spec.withStorageClassName($._config.ingester_wal_disk_class),
 
-    $.newLokiStatefulSet(name, 3, container, ingester_data_pvc) +
+  newIngesterStatefulSet(name, container, with_anti_affinity=true)::
+    $.newLokiStatefulSet(name, 3, container, [ingester_data_pvc, ingester_wal_pvc]) +
```

Comment:
IIUC, the 3 here is the number of replicas. Shouldn't this be configurable?

Reply:
The replica value is usually overwritten in a `scaling.libsonnet`, so the initial value does not matter too much - as long as it fulfils the replication factor.